test: Add Sanity Test Suite 1

dcalhoun opened the pull request:
## Related PRs

- https://github.com/WordPress/gutenberg/pull/48978

## Description

- Add Columns block placeholder test
- Add Columns block display test
- Add Columns block mover buttons test
- Add Columns block dark mode test
- Add Columns proportionate fill previews test

To test: n/a

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
test: Await modal closure for existing Columns block tests

The animated closure of the modal caused discrepancies in image
snapshots.

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
test: Add Columns block dark mode test

dcalhoun committed:
test: Add Columns block placeholder test

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
test: Reset orientation after Columns block tests

Simplify test setup for subsequent tests.

dcalhoun committed:
test: Add Columns block mover buttons test

dcalhoun committed:
test: Add Columns block display test

Verify layout spacing and alignment for portrait and landscape
orientations.

dcalhoun committed:
test: Add Columns proportionate fill previews test

dcalhoun commented:
👋🏻 @geriux. I refrained from creating new helpers so far, as I wanted to avoid abstracing prematurely. That said, I am open to any you believe are worthwhile. I'll likely re-review the tests myself seeking low-level abstractions that may make sense, but wanted to request your review now regardless. 

Let me know what you think. Thanks!

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+			await editorPage.addNewBlock( blockNames.columns );
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
There's a small typo here
```suggestion
			// Dismiss columns layout picker
```

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+			await editorPage.addNewBlock( blockNames.columns );
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
Similar as the above comment, there's a typo in the comment:
```suggestion
			// Dismiss columns layout picker
```

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+				.click();
+
+			// Click the Columns block type button
+			const blockButton = await editorPage.findBlockButton(
+				blockNames.columns
+			);
+			if ( isAndroid() ) {
+				await blockButton.click();
+			} else {
+				await editorPage.driver.execute( 'mobile: tap', {
+					element: blockButton,
+					x: 10,
+					y: 10,
+				} );
+			}
```
I wonder if we could update the `addNewBlock` helper, to be able to skip some steps if the Inserter is already visible. I mean being able to pass some options to it, so we could reuse it for this case and future blocks like `Group`, for example. What do you think?

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
```suggestion
			// Dismiss columns layout picker
```

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+			await editorPage.addNewBlock( blockNames.columns );
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
```suggestion
			// Dismiss columns layout picker
```

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
```suggestion
			// Dismiss columns layout picker
```

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+		} );
+
+		it( 'displays with correct colors with dark mode enabled', async () => {
+			if ( isAndroid() ) {
+				await editorPage.driver.executeScript( 'mobile: shell', [
+					{
+						command: 'settings put system ui_night_mode 2',
+					},
+				] );
+			} else {
+				await editorPage.driver.execute( 'mobile: setAppearance', {
+					style: 'dark',
+				} );
+			}
```
I know we've talked about having too many helpers but I think this would be a very good case for one, I think it would be used quite a bit. What do you think?

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+			await editorPage.addNewBlock( blockNames.columns );
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
+			// Dismiss coluns layout picker
```
```suggestion
			// Dismiss columns layout picker
```

geriux reviewed (commented):
Hey @dcalhoun 👋 I've left some initial comments, so exciting to see more tests added!

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+
+		// Reveal default column width sliders
+		await swipeFromTo(
+			editorPage.driver,
+			{ x: 180, y: 625 },
+			{ x: 180, y: 425 },
+			3000
```
I'm guessing these values are only for iOS right? It'd be nice if we could get the element's location instead of set values.

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,294 @@
+		expect( screenshot ).toMatchImageSnapshot();
+
+		// Dismiss block settings by tapping the modal overlay
+		const action = new wd.TouchAction( editorPage.driver );
+		action.tap( { x: 100, y: 100 } );
+		await action.perform();
```
We could also use `dismissBottomSheet`

dcalhoun committed:
refactor: Fix typo

Co-authored-by: Gerardo Pacheco <gerardo.pacheco@automattic.com>

dcalhoun committed:
test: Dismiss bottom sheet with existing helper

dcalhoun committed:
test: Use toggleDarkMode e2e mobile test helper

dcalhoun committed:
test: Dismiss bottom sheet with existing helper

dcalhoun committed:
test: Fix Android block appender button query

The available elements within this context are unexpected, with certain
elements unavailable or covered by non-visual elements, e.g. resize
observer. I consider this change a workaround. There is likely a better
way of achieving this, possibly an existing helper that could be used or
modified to support this context.

dcalhoun committed:
test: Fix Android dark mode toggling

dcalhoun committed:
test: Add initial Android snapshot images

dcalhoun committed:
test: Columns slider interactions rely upon element coordinates

Increase the stability and adaptability of the slider interaction logic.

dcalhoun committed:
fix: Remove unused import

dcalhoun committed:
test: Adjust slider test scroll position

Fix inconsistent test results on Android caused by minute differences in
final scroll position.

dcalhoun committed:
test: Decrease slider test failure threshold

Detect minute differences in column preview widths.

dcalhoun committed:
refactor: Rename for clarity

dcalhoun committed:
refactor: Leverage updated addNewBlock test helper

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
Merge branch 'trunk' of github.com:wordpress-mobile/gutenberg-mobile into test/add-sanity-test-suite-1

dcalhoun commented:
@geriux this PR and the related PR are both ready for another review. Thanks!

geriux commented on __device-tests__/gutenberg-editor-sanity-test-1-visual.test.js:
```diff
@@ -0,0 +1,307 @@
+	isEditorVisible,
+} = e2eUtils;
+
+describe( 'Gutenberg Editor - Test Suite 4', () => {
```
Let's update the name to reflect it is the correct test suite:

```suggestion
describe( 'Gutenberg Editor - Test Suite 1', () => {
```

geriux reviewed (approved):
LGTM! I left one comment related to the name of the test but overall looks good to me! Thank you for making the changes, so exciting to have more tests being added! ❤️ 

I also approved the Gutenberg PR via https://github.com/WordPress/gutenberg/pull/48978#pullrequestreview-1357168558

dcalhoun committed:
test: Fix test description typo

Co-authored-by: Gerardo Pacheco <gerardo.pacheco@automattic.com>

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
Merge branch 'trunk' of github.com:wordpress-mobile/gutenberg-mobile into test/add-sanity-test-suite-1
